src/init/cli/YargsCliExtractor.ts

```diff
@@ -3,7 +3,33 @@ import type { Arguments, Argv, Options } from 'yargs';
 import yargs from 'yargs';
 import { CliExtractor } from './CliExtractor';
 
-export type YargsArgOptions = Record<string, Options>;
+// This type exists to prevent Components.js from erroring on an unknown type
+export type YargsOptions = Options;
+
+/**
+ * This class exists as wrapper around a yargs Options object,
+ * thereby allowing us to create these in a Components.js configuration.
+ *
+ * Format details can be found at https://yargs.js.org/docs/#api-reference-optionskey-opt
+ */
+export class YargsParameter {
+  public readonly name: string;
+  public readonly options: YargsOptions;
+
+  // This line is here to prevent TS from triggering its weak type detection.
```

Comment:
But why are we preventing that?

Reply:
This is no longer needed. I had a previous version where this object extended `YargsOptions`, but forgot to remove this line when I removed it.